### PR TITLE
Preload collective page query in getInitialProps

### PR DESCRIPTION
**Pull request by znarf**
Before

Screenshot 2020-09-28 at 10 57 26

After

Screenshot 2020-09-28 at 17 27 24

### CODE REVIEW

File: pages/collective-page.js [hunk 1]
```diff
@@ -46,13 +49,31 @@ const GlobalStyles = createGlobalStyle`
...
+        query: collectivePageQuery,
+        variables: getCollectivePageQueryVariables(slug),
+      });
+      const collective = result?.data?.Collective;
+
+      // Preload Budget section if needed
+      if (collective && getFilteredSectionsForCollective(collective).includes('budget')) {
+        await client.query({
+          query: budgetSectionQuery,
+          variables: { slug, limit: 3 },
```
Comment (Betree) [inline pages/collective-page.js:73]: Would be great to have this `3` as a shared constant to make sure someone changing the number of transactions displayed will not break this improvement

**Review by Betree — APPROVED** [review]
Looks good, preloading queries like that is a good way to solve the problem for now.

In the future, having everything on GQLV2 (https://github.com/opencollective/opencollective/issues/3275) should resolve this naturally by sharing a fragment between the section and the page.

**Comment by znarf** [comment]
@Betree I'm not sure if we're going to merge that yet, I shared it to gather feedback.

**Comment by Betree** [comment]
@znarf I think it's a good improvement. We probably don't want to do that in all our pages as it introduces some complexity, but given that the collective page is the most used one and that it's pretty heavy I think the tradeoff is ok.

**Comment by znarf** [comment]
@Betree I revamped the thing to make it cleaner, and also more exhaustive. It's now doing a full preload of all potential queries. Except the first query, it will be doing it everything else in parallel which will be also a nice performance boost.

I think it's safe and ready. By saving 100ms to 500ms, it should allow most of the collective pages to be rendered under 1 sec on SSR.

**Review by Betree — APPROVED** [review]
Looking good!

**Pull request merged**